cluster_migration_core/cluster_migration_core/cluster_management/docker_framework_client.py

```diff
@@ -113,16 +141,35 @@ def remove_volume(self, volume: Volume):
         self.logger.debug(f"Removed volume {volume.name}")
 
     def create_container(self, image: str, container_name: str, network: Network, ports: List[PortMapping],
-                         volumes: List[DockerVolume], ulimits: List[Ulimit], env_variables: Dict[str, str]
+                         volumes: List[DockerVolume], ulimits: List[Ulimit], env_variables: Dict[str, str] = None,
+                         extra_hosts: Dict[str, str] = None, detach: bool = True
                          ) -> Container:
 
         # TODO - need handle if container already exists (name collision)
```

Comment:
Just wanted to note that I did go through this scenario when I restarted my PC without shutting down the HAProxy demo. `Conflict. The container name "/primary-cluster-node-1" is already in use by container`

Also got this `"could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network")` that I fixed by running `docker network prune`

Reply:
Well, sure - if you have an existing container that competes for the same name you'll have a collision.  The right place to address such collisions would be the invoking context, but the overall wrapper script that runs this stuff is a demo and isn't intended to be production-worthy.